app/controllers/shelters_controller.rb

```diff
@@ -22,7 +28,13 @@ def edit
   def update
     shelter = Shelter.find(params[:id])
     shelter.update(shelter_params)
-    redirect_to "/shelters/#{shelter.id}"
+    if shelter.save
```

Comment:
similar thing here, I believe `shelter.update` returns a boolean based on whether it successfully updated to the DB or not, so you should be able to just make this `if shelter.update(shelter_params)`